lib/chat_transcript_service.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+##
+# Used to generate BBCode [chat] tags for the message IDs provided.
+#
+# If there is > 1 message then the channel name will be shown at
+# the top of the first message, and subsequent messages will have
+# the chained attribute, which will affect how they are displayed
+# in the UI.
+#
+# Subsequent messages from the same user will be put into the same
+# tag. Each new user in the chain of messages will have a new [chat]
+# tag created.
+#
+# A single message will have the channel name displayed to the right
+# of the username and datetime of the message.
+class ChatTranscriptService
+  CHAINED_ATTR = "chained=\"true\""
+  MULTIQUOTE_ATTR = "multiQuote=\"true\""
+
+  class ChatTranscriptBBCode
+    attr_reader :channel, :multiquote, :chained
+
+    def initialize(
+      channel: nil,
```

Comment:
Should this be a required param instead? Not sure if we `channel = nil` makes sense in the context of chat.

Reply:
It's not displayed in every chat BBCode block, these are the three different displays:

![image](https://user-images.githubusercontent.com/920448/153534945-8cf56ebf-8b1c-41f3-99bc-f5652b702327.png)